StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory+FormSpec.swift

```diff
@@ -7,38 +7,111 @@
 //
 
 import Foundation
+import StripeCore
 @_spi(STP) import StripeUICore
 
 extension PaymentSheetFormFactory {
-
     func specFromJSONProvider(provider: FormSpecProvider = FormSpecProvider.shared) -> FormSpec? {
         guard let paymentMethodType = PaymentSheet.PaymentMethodType.string(from: paymentMethod) else {
             return nil
         }
         return provider.formSpec(for: paymentMethodType)
     }
 
-    func makeFormElementFromSpec(spec: FormSpec) -> FormElement {
+    func makeFormElementFromSpec(
+        spec: FormSpec,
+        additionalElements: [Element] = []
+    ) -> PaymentMethodElementWrapper<FormElement> {
         let elements = makeFormElements(from: spec)
-        return FormElement(autoSectioningElements: elements, theme: theme)
+        let formElement = FormElement(
+            autoSectioningElements: elements + additionalElements,
+            theme: theme)
+        return makeDefaultsApplierWrapper(for: formElement)
     }
 
     private func makeFormElements(from spec: FormSpec) -> [Element] {
```

Comment:
I wanted to do that, the problem is that if I want to just split out the `switch` I would either need to return a tuple with the billing details fields, where only one of them would be set or use out parameters, with the same issue.
I could do a pass on the elements array after generating them, but then I would need to check the type of the element instead of using the enum, which is possible but I try to avoid that if possible since it van be flimsy, i.e. if we change the element type, or if add a wrapper or something, it would be very easy to miss when refactoring, so I prefer switching on the enum.
In summary, I didn't find a good clean way of splitting this out, so I decided to leave it in the same function for that reason.